pkg/tide/github.go

```diff
@@ -583,3 +598,73 @@ func (mc *mergeChecker) prMergeMethod(c config.Tide, crc *CodeReviewCommon) *typ
 	}
 	return &method
 }
+
+type prowJobsByContext struct {
+	successfulBatchJob   map[string]prowapi.ProwJob
+	pendingPresubmitJobs map[string]bool
+}
+
+func getProwJobsForPRs(prs []CodeReviewCommon, pjs []prowapi.ProwJob) map[int]prowJobsByContext {
+	prowJobsForPRs := make(map[int]prowJobsByContext)
+	prNums := make(map[int]CodeReviewCommon)
+	for _, pr := range prs {
+		prNums[pr.Number] = pr
+	}
+
+	for _, pj := range pjs {
+		for _, pull := range pj.Spec.Refs.Pulls {
+			if pr, found := prNums[pull.Number]; found && pr.HeadRefOID == pull.SHA {
```

Comment:
Do we need to also consider the ref for the target branch? Could we have some race like:

1. trigger presubmits on target ref A
2. trigger batch for same
3. push to PR
4. trigger presubmits for target ref B
5. presubmits publish pending statuses
6. batch finishes
7. this code runs, overwriting presubmits 
8. batch would have been cancelled by the new push

Reply:
Okay, I see. 
I restructured this function that it checks if all PRs of a batch job are valid before collecting it. The logic is similar to how batch PRs are accumulated ([ref](https://github.com/kubernetes-sigs/prow/blob/4359601d003f3d5caf70210ff41b3768b7cbe165/pkg/tide/tide.go#L887-L916)).
Now, the context of a batch job which contain invalid or missing PRs are not collected anymore.